cypress/component/CategoricalField.cy.tsx

```diff
@@ -0,0 +1,71 @@
+import CategoricalField from '../../src/components/CategoricalField';
+
+const props = {
+  label: 'Categorical Field',
+  options: [
+    { id: '1', label: 'Option 1' },
+    { id: '2', label: 'Option 2' },
+    { id: '3', label: 'Option 3' },
+  ],
+  onFieldChange: () => {},
+  multiple: false,
+  inputValue: null,
+};
+
+describe('CategoricalField', () => {
+  it('Displays a MUI Autocomplete with the label and options passed as props', () => {
+    cy.mount(
+      <CategoricalField
+        label={props.label}
+        options={props.options}
+        onFieldChange={props.onFieldChange}
+        inputValue={props.inputValue}
+      />
+    );
+    cy.get('[data-cy="Categorical Field-categorical-field"]').should('be.visible');
+    cy.get('[data-cy="Categorical Field-categorical-field"] label').should(
+      'contain',
+      'Categorical Field'
+    );
+    props.options.forEach((option) => {
+      cy.get('[data-cy="Categorical Field-categorical-field"]').type(
+        `${option.label}{downarrow}{enter}`
+      );
+      cy.get('[data-cy="Categorical Field-categorical-field"] input').should(
+        'have.value',
+        option.label
+      );
+      cy.get('[data-cy="Categorical Field-categorical-field"]').clear();
+    });
+  });
+  it('Displays the input value passed as props', () => {
+    cy.mount(
+      <CategoricalField
+        label={props.label}
+        options={props.options}
+        onFieldChange={props.onFieldChange}
+        inputValue={props.options[0]}
+      />
+    );
+    cy.get('[data-cy="Categorical Field-categorical-field"] input').should(
+      'have.value',
+      'Option 1'
+    );
+  });
+  it('Fires onFieldChange event handler with the appropriate payload when a value is selected', () => {
+    const onFieldChange = cy.spy().as('onFieldChange');
```

Comment:
:tada: great find!
```suggestion
    const onFieldChangeSpy = cy.spy().as('onFieldChange');
```
How about that - or something to differentiate from the fixtures you are defining above.

Reply:
More generally speaking - reusing the same variable name over and over again in different context can quickly lead to confusion: "which is it now, onFieldChange the fixture, onFieldChange the prop, onFieldChange the spy, onFieldChange the alias?" Try differentiating where that is reasonably possible to make more readable